app/domain/services/prepare_clue_calculations/service.rb

```diff
@@ -16,19 +18,25 @@ def process
         # Get Responses that have not yet been used in CLUes
         # Responses with no AssignedExercise or Assignment are completely ignored
         # No order needed because of SKIP LOCKED
-        responses = Response.joins(assigned_exercise: :assignment)
-                            .where(is_used_in_clue_calculations: false)
-                            .lock('FOR NO KEY UPDATE OF "responses" SKIP LOCKED')
-                            .take(BATCH_SIZE)
+        responses = Response
+          .select(re[Arel.star], co[:ecosystem_uuid].as('latest_ecosystem_uuid'))
+          .joins(student: :course, assigned_exercise: :assignment)
+          .where(is_used_in_clue_calculations: false)
+          .lock('FOR NO KEY UPDATE OF "responses" SKIP LOCKED')
+          .take(BATCH_SIZE)
         # Get any ClueCalculations that need to be recalculated
```

Comment:
it doesn't really matter too much but I try to avoid using table names directly.  you could use `re.table_name` here I think instead of `"responses"`

Reply:
Since I have to write raw SQL anyway to get the proper lock I figured writing the table name wasn't that bad...